src/components/mobileservices/validator/rules/SameValueOfRule.test.js

```diff
@@ -0,0 +1,72 @@
+import { SameValueOfRule } from './SameValueOfRule';
+
+describe('SameValueOfRule Validation', () => {
+  it('target value is a undefined', () => {
+    const config = { target: 'app', error: 'Custom error message' };
+    const sameValueOfRule = new SameValueOfRule(config);
+
+    const input = {
+      data: { app: undefined },
+      key: 'app'
+    };
+    const value = sameValueOfRule.validate(input.data, input.key);
+    const expectResult = { valid: true };
+
+    expect(value).toEqual(expectResult);
```

Comment:
The `SameValueOfRule` is used to check that a specific field has the same value of another field. This is usually used for fields like 'password & password confirmation' or 'mail & mail confirmation'.

This test gives a positive answer, but, IMO, it doesn't test the `SameValueOfRule` _purpose_ since in the actual code, the tests is checking that `input.data.app` equals `input.data[input.data.key]; however, since `input.data.key` is `app`, it is simply checking that `input.data.app` is equals to `input.data.app`

I believe it would be better to rewrite it as something like:
```javascript
  it('target value is a undefined', () => {
    const config = { target: 'data.passwordConfirmation', error: 'Custom error message' };
    const sameValueOfRule = new SameValueOfRule(config);

    const input = {
      data: { password: undefined, passwordConfirmation: undefined },

    };
    const value = sameValueOfRule.validate(input, 'data.password');

    expect(value).toEqual({ valid: true });
  });
```